posts/deploying-a-jhipster-app-to-aws-using-elastic-beanstalk.md

```diff
@@ -0,0 +1,114 @@
+    "enableSwaggerCodegen": false,
+    "embeddableLaunchScript": false,
+    "useSass": true,
+    "clientPackageManager": "npm",
+    "clientFramework": "react",
+    "clientTheme": "none",
+    "clientThemeVariant": "",
+    "testFrameworks": [],
+    "jhiPrefix": "jhi",
+    "entitySuffix": "",
+    "dtoSuffix": "DTO",
+    "otherModules": [],
+    "enableTranslation": true,
+    "nativeLanguage": "en",
+    "languages": [
+      "en"
+    ],
+    "blueprints": []
+  }
+}
+```
+
+# Deployment to AWS with Elastic Beanstalk
+
+## First deployment
+
+First, make sure that your AWS CLI is [correctly configured](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html#cli-quick-configuration) and the account must have the `AWSElasticBeanstalkFullAccess` policy. If you use a named profile different than `default`, you just have to set the environment variable `AWS_PROFILE` with the correct profile.
+
+The command `jhipster aws` will ask questions about the AWS deployment:
+
+![](https://raw.githubusercontent.com/falydoor/blog-usa/jhipster-on-aws-with-beanstalk/images/2019/11/jhipster-aws.png)
+
+The AWS account used needs access to the S3 bucket since it will be used as a storage location for the application's Jars. The sub-generator will also create a RDS instance so make sure that the database credentials are correct and match the ones in `application-prod.yml` located in the root of your JHipster app. The application will always be packaged using the `prod` profile, which focuses on performance and scalability. Elastic Beanstalk requires a `war` and the sub-generator produces it with the command `./mvnw -ntp verify -DskipTests=true -B -Pwar,prod`. I recommend running `./mvnw clean verify` before in order to run the tests and make sure that everything is ready to go to production.
+
+Once the Elastic Beanstalk application is created, navigate to the [Elastic Beanstalk console](https://console.aws.amazon.com/elasticbeanstalk/home) to view the details of the application. Use the public URL to check that everything was correctly deployed and running.
+
+![](https://raw.githubusercontent.com/falydoor/blog-usa/jhipster-on-aws-with-beanstalk/images/2019/11/eb-url.png)
+
+If the deployment failed, the "Logs" tab on the left can be used to view all the logs of the application. For JHipster-specific issues, the logs from `/var/log/tomcat8/catalina.out` should be used to find out why the application did not start.
+
+## Application updates
+
+Updating the application is pretty simple, it is done just by running `jhipster aws` again. The sub-generator knows that there is already a Beanstalk environment and will simply update the application's code with the new version. But remember, the sub-generator will everytime package the application which means that any local changes are gonna be included. Also, the application will experience a small downtime because it has to restart (I explain how to avoid that later in this blog).
```

Comment:
```suggestion
Updating the application is pretty simple, it is done just by running `jhipster aws` again. The sub-generator knows that there is already a Beanstalk environment and will simply update the application's code with the new version. But remember, the sub-generator will package the application every time, which means that any local changes or features currently being tested will be included in the deployment. Also, the application will experience a small bit of downtime because it has to restart for the new version to be reflected (I explain how to avoid that later in this blog).
```